Ensure restfulWSClient-3.0 feature works

andymc12 opened the pull request:
For now the restfulWSClient will load everything that the server does - eventually we will split this up better.

andymc12 commented:
#build

tevans78 commented on dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.internal.restfulWS-3.0.feature:
```diff
@@ -15,7 +15,7 @@ Subsystem-Name: Internal Java RESTful Services 3.0
  io.openliberty.jakarta.validation-3.0, \
  com.ibm.websphere.appserver.org.reactivestreams.reactive-streams-1.0, \
- com.ibm.websphere.appserver.mpConfig-1.4, \
+ com.ibm.websphere.appserver.org.eclipse.microprofile.config-1.4, \
```
You already have that line

tevans78 replied:
Also, that feature depends on cdi-1.2 and tolerates 2.0 ... you're pulling in cdi-3.0

andymc12 commented:
#build

andymc12 commented:
Updated so that there is actually a difference between the `restfulWS-3.0` and `restfulWSClient-3.0` features - and added a test to verify that a RESTful client is useable with only the `restfulWSClient-3.0` feature. New #build

WhiteCat22 commented on dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/config/ConfigImpl.java:
```diff
@@ -0,0 +1,74 @@
+
+    @Override
+    public Iterable<ConfigSource> getConfigSources() {
+        new Throwable("ANDY: Config.getConfigSources").printStackTrace();
```
"ANDY"

WhiteCat22 commented on dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/config/ConfigProviderResolverImpl.java:
```diff
@@ -0,0 +1,97 @@
+
+    @Override
+    public ConfigBuilder getBuilder() {
+        new Throwable("ANDY: ConfigProviderResolverImpl.getBuilder").printStackTrace();
```
ANDY

WhiteCat22 commented on dev/io.openliberty.org.jboss.resteasy.common/src/io/openliberty/restfulWS/config/ConfigSourceImpl.java:
```diff
@@ -0,0 +1,47 @@
+    }
+    @Override
+    public String getName() {
+        new Throwable("ANDY: ConfigSourceImpl.getName").printStackTrace();
```
ANDY

andymc12 commented:
#build

andymc12 commented:
Minor tweak - new #build

andymc12 commented:
Missed the mailImpl-2.0 feature - new #build

andymc12 commented:
#run-libby-bot
